modules/Bio/EnsEMBL/Variation/Utils/VariationEffect.pm

```diff
@@ -1390,6 +1417,7 @@ sub frameshift {
     if($bvfoa->isa('Bio::EnsEMBL::Variation::TranscriptVariationAllele')) {
 
         return 0 if partial_codon(@_);
+        return 0 if stop_retained(@_);
```

Comment:
Can't it be both `frameshift` and `stop_retained`? I couldn't come up with any examples for this, so maybe it's fine.